### DISCUSSION TITLE
How to remove some edges from the original edges

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#9166] [Q&A]
For example, sometimes I want to remove the edges from the original edges.
It needs to change the `Edge_Features` and the `Edge_Index`.
It is easy to remove the original `Edge_Features` by selecting special indexes to get the new `Edge_Features`, but how to change the `Edge_Index` to match with the new `Edge_Features` rightly?

**Comment by rusty1s** [comment]
Given that you have an edge-level mask, you can simple do `edge_index = edge_index[:, mask]`. If you have indices, then you can similarly do `edge_index[:, index]`.

**Reply by StefanIsSmart** [reply]
Not Mask，
I want to remove the edges.

**Reply by rusty1s** [reply]
Yes, that's what this would do, e.g.:
```
mask = torch.ones(data.num_edges, dtype=torch.bool)
mask[index_to_remove] = False
edge_index = edge_index[:, mask]
